docs/src/utilities.md

```diff
@@ -44,12 +44,33 @@ and returns `Dict(:name=>..., :args=>..., etc.)`. The definition can be rebuilt
 calling `MacroTools.combinedef(dict)`, or explicitly with
 
 ```julia
-rtype = get(dict, :rtype, :Any)
-all_params = [get(dict, :params, [])..., get(dict, :whereparams, [])...]
-:(function $(dict[:name]){$(all_params...)}($(dict[:args]...);
-                                            $(dict[:kwargs]...))::$rtype
-      $(dict[:body])
-  end)
+rtype = get(dict, :rtype, nothing)
+wparams = get(dict, :whereparams, [])
```

Comment:
This seems unnecessarily verbose. I guess the idea is to produce nicer code in some cases, but I think the simplest thing possible is better for teaching.

Reply:
okay, so I will try to mimic the previous documentation.

Reply:
FWIW, as the splitdef author, I've found that I _always_ use `combinedef` to rebuild the definition, and I've been meaning to take out the "defintion can be rebuilt explicitly with" part entirely...

Also, `combinedef` should be exported...

Reply:
thanks for creating `splitdef`, @cstjean :)

I understand what you mean, still imho we still may want to show how to rebuild a definition expression from bare returned expressions, for a teaching purpose as @MikeInnes suggested above ?

Reply:
I think Julia will handle an empty `whereparams` list just fine. Probably worth testing to be sure, but if so we can avoid the branch.

Reply:
No, it doesn't work without the branching:
![image](https://user-images.githubusercontent.com/40514306/68474137-014e8980-0268-11ea-9d57-e84443778581.png)

Reply:
It works fine, it just doesn't print as nicely.

```julia
julia> :(f() where {} = 1)
:($(Expr(:where, :(f()))) = begin
          #= REPL[1]:1 =#
          1
      end)

julia> eval(ans)
f (generic function with 1 method)

julia> f()
1
```

Reply:
Ah yeah you're right. The example is simplified in the latest commit :)